examples/shell/shell_common/cmd_device.cpp

```diff
@@ -573,6 +573,139 @@ int cmd_device_sta(int argc, char ** argv)
     return error;
 }
 
+int cmd_device_ap(int argc, char ** argv)
+{
+    streamer_t * sout = streamer_get();
+
+    CHIP_ERROR error = CHIP_NO_ERROR;
+
+    VerifyOrExit(argc > 0, error = CHIP_ERROR_INVALID_ARGUMENT);
+
+    VerifyOrExit(PlatformMgr().TryLockChipStack(), error = CHIP_ERROR_INVALID_ARGUMENT);
+
+    if (strcmp(argv[0], "mode") == 0)
+    {
+        const char * typeStr                 = "Unknown";
+        ConnectivityManager::WiFiAPMode mode = ConnectivityMgr().GetWiFiAPMode();
+        switch (mode)
+        {
+        case ConnectivityManager::WiFiAPMode::kWiFiAPMode_NotSupported:
+            typeStr = "NotSupported";
+            break;
+        case ConnectivityManager::WiFiAPMode::kWiFiAPMode_ApplicationControlled:
+            typeStr = "ApplicationControlled";
+            break;
+        case ConnectivityManager::WiFiAPMode::kWiFiAPMode_Disabled:
+            typeStr = "Disabled";
+            break;
+        case ConnectivityManager::WiFiAPMode::kWiFiAPMode_Enabled:
+            typeStr = "Enabled";
+            break;
+        case ConnectivityManager::WiFiAPMode::kWiFiAPMode_OnDemand:
+            typeStr = "OnDemand";
+            break;
+        case ConnectivityManager::WiFiAPMode::kWiFiAPMode_OnDemand_NoStationProvision:
+            typeStr = "OnDemand-NoStation";
+            break;
+        }
+        streamer_printf(sout, "%s\r\n", typeStr);
+    }
+    else if (strcmp(argv[0], "set_mode") == 0)
+    {
+        if (argc < 2)
+        {
+            streamer_printf(sout,
+                            "Invalid command: needs specify mode:\r\n"
+                            "0: Disable\r\n"
```

Comment:
How about using strings instead?

`set_mode softap_enable`

is more readable than

`set_mode 1`

Reply:
Acked